### ISSUE TITLE
add unit tests for handling of LFNBase and forceMerged in workflow creation/assignment

### ISSUE THREAD

**Issue by hufnagel** [dmwm/WMCore#8016]
Make sure that when a workflow is created with a non-standard LFNBase, that non-standard LFNBase survives assignment.

Make sure when a workflow has a task with forceMerged output, that output stays forceMerged after assignment.

These are basically the two things that were broken for the Tier0 and the same reason they broke now (these features aren't really used in WMCore) could cause them to be broken again later. A unit test specifically for this should prevent this.

I would look into it myself, but I have too much already on my plate and @amaltaro it seems to me you just recently looked at this code.

**Comment by amaltaro** [comment]
by "non-standard" you mean using the value specified during creation and not the default value?

Which spec files are we talking about?

Can you provide the dictionary passed during assignment for each T0 spec file that you want to have it tested?

**Comment by hufnagel** [comment]
Yes, LFNBase is set to something different than default during creation and the forceMerged = True is passed from within the spec (Express spec, maybe we can add a test in the Express spec unit test ?).

The calls the Tier0 uses to do the "assignment" are here:

https://github.com/dmwm/T0/blob/master/src/python/T0/RunConfig/RunConfigAPI.py#L676
https://github.com/dmwm/T0/blob/master/src/python/T0/RunConfig/RunConfigAPI.py#L1075

Pretty basic, just want to make sure that these type of calls don't automatically override LFNBase and the forceMerged setting.

**Comment by hufnagel** [comment]
Any news on this ?

**Comment by amaltaro** [comment]
Not yet, these output data checks are annoying and not a 15min task.
I might dedicate some time for it in the coming week though
